tool/src/grapheme_category_loader.dart

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) 2018, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import "dart:async";
+import "dart:io" show stderr;
+import "dart:typed_data";
+
+import "package:characters/src/grapheme_clusters/constants.dart";
+
+import "data_files.dart";
+
+// Loads the grapheme breaking categories from Unicode data files.
+
+Future<Uint8List> loadGraphemeCategories(
+    {bool update = false, bool verbose = false}) async {
+  List<String> dataFiles = await Future.wait([
+    graphemeBreakPropertyData.load(checkForUpdate: update),
+    emojiData.load(checkForUpdate: update),
+    // This data used to be in:
+    // https://www.unicode.org/Public/12.0.0/ucd/auxiliary/GraphemeBreakProperty-12.0.0d16.txt
+    Future.value(
+        "D800..DFFF    ; Control # Cc       <control-D800>..<control-DFFF>\n"),
+  ]);
+  var table = _parseCategories(dataFiles, verbose: verbose);
+  return table;
+}
+
+// -----------------------------------------------------------------------------
+// Unicode table parser.
+final _tableRE = RegExp(r"^([\dA-F]{4,5})(?:..([\dA-F]{4,5}))?\s*;\s*(\w+)\s*#",
+    multiLine: true);
+
+// The relevant names that occur in the Unicode tables.
+final categoryByName = {
+  "CR": categoryCR,
+  "LF": categoryLF,
+  "Control": categoryControl,
+  "Extend": categoryExtend,
+  "ZWJ": categoryZWJ,
+  "Regional_Indicator": categoryRegionalIndicator,
+  "Prepend": categoryPrepend,
+  "SpacingMark": categorySpacingMark,
+  "L": categoryL,
+  "V": categoryV,
+  "T": categoryT,
+  "LV": categoryLV,
+  "LVT": categoryLVT,
+  "Extended_Pictographic": categoryPictographic,
+};
+
+Uint8List _parseCategories(List<String> files, {required bool verbose}) {
+  var result = Uint8List(0x110000);
+  result.fillRange(0, result.length, categoryOther);
+  int count = 0;
+  var categoryCount = <String, int>{};
+  var categoryMin = <String, int>{
+    for (var category in categoryByName.keys) category: 0x10FFFF
+  };
+  int min(int a, int b) => a < b ? a : b;
```

Comment:
Import from `dart:math`?

Reply:
That one is wasteful because it works for `double` too.